src/gluonts/torch/model/estimator.py

```diff
@@ -197,6 +198,8 @@ def train_model(
         custom_callbacks = self.trainer_kwargs.get("callbacks", [])
         callbacks = [checkpoint] + custom_callbacks
         trainer_kwargs = {**self.trainer_kwargs, "callbacks": callbacks}
+
+        trainer_kwargs = matching_arguments(pl.Trainer, trainer_kwargs)
```

Comment:
I'm wondering: is this desirable? Someone providing the wrong `kwargs` may not realize they are filtered out, while they should error

Reply:
oh I see what's going on: `patience` needs to be filtered out... maybe then `patience` doesn't really belong to `trainer_kwargs`

Reply:
Since LR scheduling is a part of model training, ideally, it should be an option for `Trainer` function. The interface to interact with training functionality in Pytorch estimators is via `trainer_kwargs`, so I put the option of setting `patience` here.

Reply:
Also, I just realized that Pytorch `DeepAREstimator` doesn't provide any interface to set learning rate explicitly by the user. So, maybe we can expose `lr` in `trainer_kwargs` as well or something else. Any suggestions are welcome.

Reply:
Sure, however this contrasts a bit with what the `DeepAREstimator` docstring says:

```
trainer_kwargs
        Additional arguments to provide to ``pl.Trainer`` for construction.
```

In fact, there we put other training-related arguments like batch size as separate. I think the docstring above should really say `Arguments to pl.Trainer for construction` and be in this class. And then the role of `trainer_kwargs` is clear and there should be no special cases.

So, I vote for making `patience` an explicit argument to the constructor :-)

Reply:
> So, maybe we can expose `lr` in `trainer_kwargs` as well or something else. Any suggestions are welcome.

Good spot! Same as for patience, I would keep it a separate argument, and let trainer_kwargs feed directly into pl.Trainer.

However I would address the `lr` issue in a separate PR since its technically a bug fix that we would want to backport.